eng/pipelines/templates/steps/vcpkg.yml

```diff
@@ -12,15 +12,14 @@ steps:
         )
     displayName: vcpkg --version
 
-  - ${{if and(eq(variables['System.TeamProject'], 'internal'), ne(variables['Build.Reason'], 'PullRequest')) }}:
```

Comment:
This pipeline is used in both public and internal. Since public doesn't have access to the artifacts subscription this step will always fail if that condition is removed. The condition should be modified to only run on internal projects, meaning that the PR check should be removed: 

```yaml
- ${{if eq(variables['System.TeamProject'], 'internal') }}:
```

That will exclude the step from public pipeline runs and keep it in internal runs.

The tradeoff here is that it's possible to update the cache from a PR by commenting `/azp run <pipeline>`. I think we initially avoided updating this cache from PRs. Since authorized action is still required to update the cache, I think this is OK.

Reply:
Thank you for the explanation, Daniel!
To other readers: before making this change, we talked with Daniel, he proposed this change, and after I added it to this PR, he left the comment above to clarify it for other PR reviewers.